packages/core/src/state-management/state-manager.ts

```diff
@@ -192,6 +192,7 @@ export class StateManager {
   }
 
   publishTip(state$: RunningState): void {
+    // NOTE: pass model to publishTip
     this.dispatcher.publishTip(state$.id, state$.tip)
```

Comment:
Something like this, naming is likely off.
```suggestion
    // NOTE: pass model to publishTip
    this.dispatcher.publishTip(state$.id, state$.tip, state$.state.metadata.model)
```